.design/requirements-575.md

```diff
@@ -0,0 +1,269 @@
+* *Partial* message handlers are notified by the implementation when it is ready to deliver frames of a whole message.
+
+As it is explicitly stated in the JavaDoc part of the specification, the *partial* message handler is implementation-dependent.
+As a result, the OSGi WebSocket Whiteboard specification should reuse the `MessageHandler.Whole` type but ignore the `MessageHandler.Partial` one and the methods that accept it as argument.
+
+## Use Cases
+
+### UC1: Define and register a WebSocket server endpoint using annotations
+
+The code below is a WebSocket handler that we would want to register on the `/ws/work/{targetId}` endpoint.
+
+The OSGi WebSocket Whiteboard implementation is responsible to register an instance of this service as WebSocket endpoint and message handler, either directly or using a proxy endpoint class.
+
+```java
+@Component(service = WebSocketEndpointExample.class, scope = ServiceScope.PROTOTYPE)
+@WebSocketWhiteboardServletPattern("/ws/work/{targetId}") // See "WebSocket pattern description"
+public class WebSocketEndpointExample {
+    @OnOpen
+    public void onOpen(
+        @PathParam("targetId") String targetId,
+        Session session) {
+        // WebSocket stream is opened
+    }
+
+    @OnClose
+    public void onClose(
+        @PathParam("targetId") String targetId,
+        Session session,
+        CloseReason reason) {
+        // WebSocket stream is closed
+    }
+
+    @OnError
+    public void onError(
+        @PathParam("targetId") String targetId,
+        Session session,
+        Throwable error) {
+        // Error handling the WebSocket stream
+    }
+
+    @OnMessage
+    public void onMessage(
+        @PathParam("targetId") String targetId,
+        String message,
+        Session session) {
+        // Process incoming message
+		// The "message" argument can also be of type Reader, Decoder.Text
+		// or Decoder.TextStream.
+		// See @OnMessage documentation for more details.
+    }
+
+	@OnMessage
+    public void onMessage(
+        @PathParam("targetId") String targetId,
+        byte[] message,
+        Session session) {
+        // Process binary message
+		// The "message" argument can also be of type ByteBuffer, InputStream,
+		// Decoder.Binary or Decoder.BinaryStream.
+		// See @OnMessage documentation for more details.
+    }
+
+    @OnMessage
+    public void onPong(
+        @PathParam("targetId") String targetId,
+        PongMessage message,
+        Session session) {
+        // Process pong message
+    }
+}
+```
+
+#### WebSocket pattern description
+
+In a Jakarta environment, we would annotate this class with `@ServerEndpoint("/ws/work/{targetId}")` and register it in a Jakarta WebSocket-ready HTTP server.
+
+In OSGi, we might want to use annotations similar to the ones in the Jarkata Servlet whiteboard specification to provide more control, *e.g.* `@WebSocketWhiteboardServletPattern("/ws/work/{targetId}")`.
```

Comment:
Why? While a servlet *might* be one way to implement this, it is not mandatory and the `@ServerEndoint` carry enough information already.